### DISCUSSION TITLE
how to generate and store a key pair?

### DISCUSSION THREAD

**Discussion by koo5** [panva/jose#150] [Q&A]
your documentation states:
```
const privateKey = await parseJwk({
  alg: 'ES256',
  crv: 'P-256',
  kty: 'EC',
  d: 'VhsfgSRKcvHCGpLyygMbO_YpXc7bVKwi12KQTE4yOR4',
  x: 'ySK38C1jBdLwDsNWKzzBHqKYEE5Cgv-qjWvorUXk9fw',
  y: '_LeQBw07cf5t57Iavn4j-BqJsAD1dpoz8gokd3sBsOo'
})
```
however, i cannot find anything on how to arrive at this "Jwk". When i follow the example on https://github.com/panva/jose/blob/main/docs/functions/_util_generate_key_pair_.generatekeypair.md#readme ,
the output is :
```
PublicKeyObject { [Symbol(kKeyType)]: 'public' }
PrivateKeyObject { [Symbol(kKeyType)]: 'private' }
```

**Comment by mhamo-mesdocteurs** [comment]
Not easy to find that info, this could be in the FAQ

```
import generateKeyPair from 'jose/util/generate_key_pair';
import fromKeyLike from 'jose/jwk/from_key_like';

{...}

const keyPair = await generateKeyPair(alg, {
  modulusLength: 2048, // default value
});

const privateJwk = await fromKeyLike(keyPair.privateKey);
const publicJwk = await fromKeyLike(keyPair.publicKey);
```

**Reply by panva** [reply]
Why FAQ when all of this is in the documentation. All the modules that do this are described by what they do. Furthermore the [KeyLike](https://github.com/panva/jose/blob/main/docs/types/_types_d_.keylike.md) type links to platform specific means of obtaining KeyLike objects as well.

<img width="395" alt="Screenshot 2021-01-22 at 12 31 52" src="https://user-images.githubusercontent.com/241506/105485917-e8d06080-5cad-11eb-9a1d-22be9dff84e0.png">

**Comment by mhamo-mesdocteurs** [comment]
Comming from `jose@2` where we can generate a `jwk` with only a function

```
import { JWK } from 'jose';

{...}

const key = await JWK.generate(kty, alg, { use: 'sig' });
```

On `jose@3` I was looking for a similar function. Or at least a code sample to help me in the migration.

I didn't look `Parsing (JWK to KeyLike)` and `Conversion (KeyLike to JWK)` because from what I understood I didn't know how to get the `KeyLike`.

I think that it's not very explicit that `Asymmetric Key Pair Generation` is the way to generate a `KeyLike`
